src/app/webserver/webserver.component.ts

```diff
@@ -79,4 +89,40 @@ export class WebServerComponent {
     get service() {
         return this._service;
     }
+
+    get server(): Promise<WebServer> {
+        return new Promise<WebServer>((resolve, reject) => {
+            this._service.server.catch(e => {
+                if (e instanceof UnexpectedServerStatusError) {
+                    this._notifications.confirm(
+                        `Restart IIS Administration API`,
+                        `IIS Administration API is currently ${e.Status}, restart?`).then(confirmed => {
+                        if (confirmed) {
+                            var sub = this._runtime.RestartIISAdministration().subscribe(
+                                _ => {
+                                    this._service.server.catch(e => {
+                                        reject(this.failure = `Unable to restart IIS Administration API Service`)
+                                        throw e
+                                    }).then(s => {
+                                        resolve(s)
+                                    })
+                                },
+                                _ => {
+                                    reject(this.failure = `Unable to restart IIS Administration API Service`)
+                                },
+                                () => { sub.unsubscribe() },
+                            )
+                        } else {
+                            reject(this.failure = `Web Server Module cannot be initialized. IIS Administration API status: ${e.Status}`)
+                        }
+                    })
+                } else {
+                    reject(this.failure = `Unknown error has occurred when trying to initialize Web Server Module`)
```

Comment:
I think we should expose the detailed error information for easy troubleshoot.

Reply:
Added basic errors. It would still not be quite readable but user should be able to refer to the console logs for more details